vsphere/datadog_checks/vsphere/legacy/event.py

```diff
@@ -30,10 +30,11 @@
     'VmPoweredOnEvent': [],
     'VmPoweredOffEvent': [],
     'VmReconfiguredEvent': [],
-    'VmResumedEvent': [],
     'VmSuspendedEvent': [],
 }
 
+ALLOWED_EVENTS = [getattr(vim.event, event_type) for event_type in EXCLUDE_FILTERS.keys()]
```

Comment:
Should this be on the legacy part of the integration?

Reply:
Yes, event collection has not been modified with the revamp. The logic is shared by both implementations

Reply:
If its shared, and this is not for this PR, I think it should not be inside the legacy part

Reply:
That's a valid point, in that case we should duplicate the events collection logic

Reply:
or extract it to the non legacy part. The legacy part can import from outside, right?

Reply:
That's technically possible yes.

But from my perspective I understand the concern that some code is shared by two implementations and we can't improve the events collection logic for just the new implementation without also changing the legacy implementation. I don't think the location of that `event.py` file matter that much though wdyt?

Reply:
I just want to _one day_ safely be able to delete the legacy folder

Reply:
Yes, agreed for preparing the field so we can delete legacy easily when needed. 

+1 for moving code still used in non-legacy folders (in a separate PR).